gufe/protocols/protocolunit.py

```diff
@@ -0,0 +1,78 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+
+"""The `ProtocolUnit` class should be subclassed for all units to be used as
+part of a `ProtocolDAG`.
+
+"""
+
+import abc
+from os import PathLike
+from typing import Iterable, List, Dict, Any, Optional
+
+from .results import ProtocolUnitResult
+
+
+class ProtocolUnit(abc.ABC):
+    """A unit of work computable by"""
+
+    def __init__(
+        self,
+        settings: Optional["ProtocolSettings"] = None,  # type: ignore
+        name: Optional[str] = None,
+        **kwargs
+    ):
+
+        self._settings = settings
+        self._kwargs = kwargs
+        self._name = name
+
+    def __hash__(self):
```

Comment:
Should there be an `__eq__` to deduplicate units? (or maybe deduplication isn't happening at this level?)